oggm/utils.py

```diff
@@ -1430,6 +1431,18 @@ def _divide_task(gdir, div_id=None, **kwargs):
         return _divide_task
 
 
+def self_iterating_entity_task(task_func):
+    """
```

Comment:
Yes, the name is indeed a bit clunky.
Didn't know about them being called global_tasks, calling the decorator global_task seems way better.